ssz/src/main/java/tech/pegasys/teku/ssz/impl/AbstractSszMutableComposite.java

```diff
@@ -56,84 +51,139 @@
 
   protected AbstractSszComposite<SszChildT> backingImmutableData;
   private Consumer<SszMutableData> invalidator;
-  private final Map<Integer, SszChildT> childrenChanges = new HashMap<>();
-  private final Map<Integer, SszMutableChildT> childrenRefs = new HashMap<>();
-  private final Set<Integer> childrenRefsChanged = new HashSet<>();
+  private final Map<Integer, ChildChangeRecord<SszChildT, SszMutableChildT>> childrenChanges =
+      new HashMap<>();
   private Integer sizeCache;
+  private final SszCompositeSchema<?> cachedSchema;
+
+  private static final class ChildChangeRecord<
```

Comment:
(nit) Usually we put internal classes at the bottom of the file

Reply:
Done: b8924546